src/Raven.Server/Smuggler/Documents/SmugglerImporter.cs

```diff
@@ -383,18 +383,16 @@ public override void Execute(DocumentsOperationContext context, RavenTransaction
                     if (metadata.TryGet(Constants.Metadata.Id, out key) == false)
                         throw new InvalidOperationException("Document's metadata must include the document's key.");
 
-                    DynamicJsonValue mutatedMetadata;
-                    metadata.Modifications = mutatedMetadata = new DynamicJsonValue(metadata);
-                    mutatedMetadata.Remove(Constants.Metadata.Id);
-                    mutatedMetadata.Remove(Constants.Metadata.Etag);
+                    document.PrepareForStorage();
+                    BlittableJsonReaderObject modifiedDocument = context.ReadObject(document, key);
```

Comment:
This shouldn't be done here. This should be done outside of the command.
The reason is that we want to reduce as much as possible the amount of work that we are doing here while holding the tx.